lib/vector-core/proto/event.proto

```diff
@@ -35,10 +35,12 @@ message Log {
   // Deprecated, use value instead
   map<string, Value> fields = 1;
   Value value = 2;
+  Value metadata = 3;
 }
 
 message Trace {
   map<string, Value> fields = 1;
+  Value metadata = 3;
```

Comment:
Is there a specific reason why this is `3` instead of `2`?

Reply:
I'd have to brush up on my Protocol Buffers knowledge, but I'm not sure we could ever actually make a change such that we switched from using the `Trace` message type to `Log` without it being a backwards-incompatible change, and so aligning the field IDs probably doesn't matter at all here.

It also just seems unlikely we'd ever do that, because you're generally limited in terms of being able to do newtype wrapper tricks, etc. The Protocol Buffers definition should be explicit, and separate from whatever tricks we do internally.

I originally approved the PR because using 2 vs 3 doesn't materially matter from a parsing standpoint, but we probably _should_ just have it as 2 for lack of misunderstandings around _why_ it's skipping a number.

Reply:
I'll just switch it to 2. As @tobz says, it doesn't materially matter, but it's clearly confusing since this discussion is happening.